### PR TITLE
Azure compute plugin three

### PR DESCRIPTION
**Pull request by fgan1**
## Description
Implementing Azure Compute Plugin

## Note
- It's not the final version. I didn't do the integration tests. Because of this, the PR will be merged in the azure-compute-plugin branch instead of develop branch.
- The Azure SDK uses RXJava(reactive programming).

## Proposed changes
- Implement delete instance method in the Azure Compute plugin
- Create some utility classes to the Azure context

## Additional information
This branch depends on a branch in the Fogbow Common (azure-compute-plugin-one)

## PR Slices
[x] One: Get instance context
[x] Two: Request instance context
[x] Three: Delete instance context

## Reminding
PRs order:
azure-compute-plugin < PR1 < *PR2 < PR3*

### CODE REVIEW

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/sdk/AzureVirtualMachineOperationSDKTest.java [hunk 1]
```diff
@@ -397,9 +400,10 @@ public void testBuildAzureVirtualMachineObservable() throws InvalidParameterExce
 
         PowerMockito.mockStatic(AzureNetworkSDK.class);
         NetworkInterface networkInterfaceExcepted = Mockito.mock(NetworkInterface.class);
+        Optional<NetworkInterface> networkInterfaceExceptedOptional = Optional.ofNullable(networkInterfaceExcepted);
```
Comment (jadsonluan) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/sdk/AzureVirtualMachineOperationSDKTest.java]: I think you misspelled "expected"

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureNetworkSDKTest.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        Optional<NetworkInterface> networkInterface =
+                AzureNetworkSDK.getNetworkInterface(azure, networkInterfaceId);
+
+        // verify
+        Assert.assertTrue(networkInterface.isPresent());
+        Assert.assertEquals(networkInterfaceExpected, networkInterface.get());
+    }
+
+    // test case: When calling the getNetworkInterface method and do not find a network interface,
+    // it must verify if It returns a Optional with a network interface.
```
Comment (jadsonluan) [inline src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureNetworkSDKTest.java]: The test case is misspelled

**Current:** `// it must verify if It returns a Optional with a network interface.`
**Right:** `// it must verify if It returns a Optional without a network interface.`

File: src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureNetworkSDK.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        try {
+            NetworkInterfaces networkInterfaces = getNetworkInterfaces(azure);
+            NetworkInterface networkInterface = networkInterfaces.getById(azureNetworkInterfaceId);
+            return Optional.ofNullable(networkInterface);
+        } catch (RuntimeException e) {
+            throw new UnexpectedException(e.getMessage(), e);
+        }
+    }
+
+    public static NetworkInterfaces getNetworkInterfaces(Azure azure) {
```
[the remark this answers is not in the document]
Comment (fgan1) [reply src/main/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureNetworkSDK.java]: Done.
I've made the change in the branch azure-compute-plugin-four-pr-fixes

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/sdk/AzureComputePluginTest.java [hunk 4]
```diff
@@ -301,4 +301,26 @@ public void testBuildComputeInstanceSuccessfully() {
...
+    @Test
+    public void testDeleteInstanceSuccessfully() throws FogbowException {
+        // set up
+        ComputeOrder computeOrder = Mockito.mock(ComputeOrder.class);
+        String instanceId = "instanceId";
+        Mockito.when(computeOrder.getInstanceId()).thenReturn(instanceId);
+
+        Mockito.doNothing()
+                .when(this.azureVirtualMachineOperation)
+                .doDeleteInstance(Mockito.any(), Mockito.any());
```
[the remark this answers is not in the document]
Comment (fgan1) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/sdk/AzureComputePluginTest.java]: Done.
I've made the change in the branch azure-compute-plugin-four-pr-fixes

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/sdk/AzureVirtualMachineOperationSDKTest.java (hunk 1 above)
Comment (fgan1) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/compute/sdk/AzureVirtualMachineOperationSDKTest.java]: Done.
I've made the change in the branch azure-compute-plugin-four-pr-fixes

File: src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureNetworkSDKTest.java (hunk 2 above)
Comment (fgan1) [reply src/test/java/cloud/fogbow/ras/core/plugins/interoperability/azure/network/sdk/AzureNetworkSDKTest.java]: Done.
I've made the change in the branch azure-compute-plugin-four-pr-fixes

**Pull request merged**